front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/InfrastructureDefinitionBucketDAO.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.model;
+
+import com.netflix.spinnaker.front50.exception.NotFoundException;
+import com.netflix.spinnaker.front50.model.infrastructureDefinition.InfrastructureDefinition;
+import com.netflix.spinnaker.front50.model.infrastructureDefinition.InfrastructureDefinitionDAO;
+import org.springframework.util.Assert;
+import rx.Scheduler;
+
+import java.util.Collection;
+import java.util.UUID;
+
+public class InfrastructureDefinitionBucketDAO extends BucketDAO<InfrastructureDefinition> implements InfrastructureDefinitionDAO {
+
+    public InfrastructureDefinitionBucketDAO(String basePath,
+                             StorageService service,
+                             Scheduler scheduler,
+                             int refreshIntervalMs) {
+        super(InfrastructureDefinition.class, "infrastructureDefinitions",
+                basePath, service, scheduler, refreshIntervalMs);
+    }
+
+    @Override
+    public Collection<InfrastructureDefinition> getInfrastructureDefinitionHistory(String application, String account, int maxResults) {
+        String id = getInfrastructureDefinitionId(application, account);
+        return allVersionsOf(id, maxResults);
+    }
+
+    @Override
+    public String getInfrastructureDefinitionId(String application, String account) {
```

Comment:
What are the expectations around search?

Something along the lines of ... "Show me what things looked like at time T?"